app/support/products/scheduling_support.rb

```diff
@@ -94,12 +94,16 @@ def available_schedule_rules(user)
     end
   end
 
-  def online!
-    offline_reservations.current.update_all(reserve_end_at: Time.current)
+  def offline?
+    offline_reservations.current.any?
+  end
+
+  def offline_via_schedule_share?
+    schedule.products.any?(&:offline?)
   end
 
-  def online?
-    offline_reservations.current.none?
+  def online!
+    offline_reservations.current.update_all(reserve_end_at: Time.current)
```

Comment:
I don't think there are any downsides to not running validations in this case, are there?